src/driver-hidpp20.c

```diff
@@ -54,6 +54,7 @@
 #define HIDPP_CAP_KBD_REPROGRAMMABLE_KEYS_1b00		(1 << 4)
 #define HIDPP_CAP_COLOR_LED_EFFECTS_8070		(1 << 5)
 #define HIDPP_CAP_ONBOARD_PROFILES_8100			(1 << 6)
+#define HIDPP_CAP_REMOVABLE_RESOLUTIONS			true // I don't get this definition system here...
```

Comment:
So, the way that this works is that we have a variable `capabilities` and each bit of that variable represents a specific capability. Here we define the bit that represents each capability. 1 is `b00000001` so if we do `1 << 4` it becomes `b00010000`.

Then you can do:
```
if (capabilities & HIDPP_CAP_REMOVABLE_RESOLUTIONS)
	/* The device supports removable resolutions! So let's do something interesting... */
```
If the 5th bit from the right is set in `capabilities`, the condition will be triggered.

I am not sure you need a HID++ 2.0 capability to implement what you want, but feel free to use it to make progress. It's better to do this step by step.

Reply:
Ahhh thanks I recognize it know, it's the bit shift operator.